pilot/pkg/model/service.go

```diff
@@ -1313,19 +1313,29 @@ func (s *Service) GetAllAddressesForProxy(node *Proxy) []string {
 }
 
 func (s *Service) getAllAddressesForProxy(node *Proxy) []string {
+	addresses := []string{}
 	if node.Metadata != nil && node.Metadata.ClusterID != "" {
-		addresses := s.ClusterVIPs.GetAddressesFor(node.Metadata.ClusterID)
-		if (features.EnableDualStack || features.EnableAmbient) && node.GetIPMode() == Dual {
-			return addresses
+		addresses = s.ClusterVIPs.GetAddressesFor(node.Metadata.ClusterID)
+	}
+	if len(addresses) == 0 && bool(node.Metadata.DNSCapture) &&
+		(bool(node.Metadata.DNSAutoAllocate) || features.EnableIPAutoallocate) {
+		// maybe autoallocated
+		if s.AutoAllocatedIPv4Address != "" {
+			addresses = append(addresses, s.AutoAllocatedIPv4Address)
 		}
-		addresses = filterAddresses(addresses, node.SupportsIPv4(), node.SupportsIPv6())
-		if len(addresses) > 0 {
-			return addresses
+		if s.AutoAllocatedIPv6Address != "" {
```

Comment:
Should this be an `else if` or is it possible for both cases to be true?

Reply:
The goal is to provide better support dual-stack so if both autoIPs are set we collect both and allow the filtering below to sort it out if needed.